### ISSUE TITLE
Glyph development stage over time visual

### ISSUE THREAD

**Issue by chrissimpkins** [googlefonts/fontelemetry#19]
Question to be addressed: What is the stage of glyph development over time across specified design stages?

Request from Dave Crossland for a ~~heatmap~~ area line chart that displays:

- x-axis : time
- y-axis: % complete

**Comment by davelab6** [comment]
Yes - we chatted on a video call about it, looking forward to next sketches :)

**Comment by chrissimpkins** [comment]
Here is what I am thinking.  This is laid out as % on the y-axis with a line for each stage.  The area is stacked to 100% and starts at 100% "Begin" stage, eventually progresses to 100% "Done" stage over time as work occurs with intermediate stages up and down based upon analysis of each glyph's stage.  This could also use glyph count on the y-axis.  Presentation would not differ.

![4l6wg-video](https://user-images.githubusercontent.com/4249591/52667464-eae4ee80-2ede-11e9-9437-7ad89e188bb1.gif)

**Comment by chrissimpkins** [comment]
It is possible to get the git commit of a repository at a given date using `git rev-list` or `git rev-parse`.  The latter has a simpler syntax but only preserves 90 days of commits in the history.  The former, as I understand it, can be used against the entire git history.

References: 
- https://stackoverflow.com/a/6990682
- https://git-scm.com/docs/git-rev-list

```
git checkout $(git rev-list -n 1 --first-parent --before="2009-07-27 13:37" master)
```

**Comment by alexeiva** [comment]
`y-axis` may include additional data from git status, such as:
`rewrite documents folder/file.glyphs (97%)` # how much of the document has been changed — is it a minor or major update.
`41497 insertions (+) / 1726 deletions (-)` # by this we can calculate how many nodes were touched

**Comment by chrissimpkins** [comment]
Thanks Alexei! Let me think about how we could approach something like that in the figure.  Will push more examples here when they are available.

**Comment by chrissimpkins** [comment]
Weekly historical git checkout of source from current date / time will look something like this:

```python
import datetime
import shlex
import subprocess

d = datetime.date.today()
for weeksback in range(1, 52):
    dd = d - datetime.timedelta(weeks=weeksback)
    git_command = shlex.split("git checkout $(git rev-list -n 1 --first-parent --before='{}' master)".format(dd))
    try:
        subprocess.run(git_command, capture_output=True)
    except Exception as e:
        sys.stderr.write(str(e))
        sys.exit(1)

    # perform the data pull from the source files at this state of the git version control history
    # ...

```

If this code reaches the end of the git history in time, the git rev-list command returns an empty string with zero exit status code.

**Comment by chrissimpkins** [comment]
@jeremiehornus @kalapi Do you have several weeks of mark color data in place that we can use to test this?  The source to pull from Glyphs source files is in place and ready to go.  Source to pull color data from UFO files should be ready to go in the next week.

**Comment by chrissimpkins** [comment]
The source is in place to support mark color data pulls from UFO source files (in addition to glyphs source files) as of https://github.com/googlefonts/fontelemetry/commit/c30ac03568efab22d16e9108b7cac3901c56e6bd .

**Comment by chrissimpkins** [comment]
I have source in place to get to counts ✕ color definition ✕ time.  Here is an example from an actual body of UFO typeface source. (cc @jeremiehornus)

Current

```
Status 1: 828 (86.79%)
Status 2: 71 (7.44%)
Status 3: 31 (3.25%)
White: 24 (2.52%)
```


1 week prior:

```
Status 1: 828 (86.79%)
Status 2: 71 (7.44%)
Status 3: 31 (3.25%)
White: 24 (2.52%)
```

2 weeks prior:

```
Status 1: 801 (84.76%)
Status 2: 71 (7.51%)
Status 3: 49 (5.19%)
White: 24 (2.54%)
```

3 weeks prior:

```
Status 1: 573 (44.38%)
Status 2: 570 (44.15%)
Status 3: 124 (9.6%)
White: 24 (1.86%)
```

4 weeks prior:

```
Status 1: 496 (39.02%)
Status 2: 570 (44.85%)
Status 3: 181 (14.24%)
White: 24 (1.89%)
```
